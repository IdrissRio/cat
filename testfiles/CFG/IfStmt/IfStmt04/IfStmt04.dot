digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "InstanceInitializer[28]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Block[29]"}
{ rank = same; "List[30]"}
{ rank = same; "VarDeclStmt[31]"; "VarDeclStmt[40]"; "IfStmt[49]"; "VarDeclStmt[85]"}
{ rank = same; "Modifiers[32]"; "PrimitiveTypeAccess[34]"; "List[35]"; "Modifiers[41]"; "PrimitiveTypeAccess[43]"; "List[44]"; "VarAccess[50]"; "Block[51]"; "Opt[84]"; "Modifiers[86]"; "PrimitiveTypeAccess[88]"; "List[89]"}
{ rank = same; "List[33]"; "VariableDeclarator[36]"; "List[42]"; "VariableDeclarator[45]"; "List[52]"; "List[87]"; "VariableDeclarator[90]"}
{ rank = same; "List[37]"; "Opt[38]"; "List[46]"; "Opt[47]"; "IfStmt[53]"; "VarDeclStmt[76]"; "List[91]"; "Opt[92]"}
{ rank = same; "BooleanLiteral[39]"; "BooleanLiteral[48]"; "VarAccess[54]"; "Block[55]"; "Opt[65]"; "Modifiers[77]"; "PrimitiveTypeAccess[79]"; "List[80]"}
{ rank = same; "List[56]"; "Block[66]"; "List[78]"; "VariableDeclarator[81]"}
{ rank = same; "VarDeclStmt[57]"; "List[67]"; "List[82]"; "Opt[83]"}
{ rank = same; "List[59]"; "VariableDeclarator[62]"; "Modifiers[69]"; "PrimitiveTypeAccess[71]"; "List[72]"}
{ rank = same; "Modifiers[58]"; "PrimitiveTypeAccess[60]"; "List[61]"; "VarDeclStmt[68]"}
{ rank = same; "List[74]"; "Opt[75]"}
{ rank = same; "List[63]"; "Opt[64]"; "List[70]"; "VariableDeclarator[73]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < IfStmt04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitializer[28]" [label="InstanceInitializer\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[29]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[31]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[32]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[34]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[35]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[36]" [label="VariableDeclarator\n < b >", fillcolor=white   style=filled  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[38]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[39]" [label="BooleanLiteral\n < true >", fillcolor=white   style=filled  ]
"VarDeclStmt[40]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[41]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[43]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[45]" [label="VariableDeclarator\n < a >", fillcolor=white   style=filled  ]
"List[46]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[47]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[48]" [label="BooleanLiteral\n < true >", fillcolor=white   style=filled  ]
"IfStmt[49]" [label="IfStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[50]" [label="VarAccess\n < b >", fillcolor=white   style=filled  ]
"Block[51]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[52]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IfStmt[53]" [label="IfStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[54]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"Block[55]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[56]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[57]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[58]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[59]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[60]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[61]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[62]" [label="VariableDeclarator\n < c >", fillcolor=white   style=filled  ]
"List[63]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[64]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[65]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[66]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[67]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[68]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[69]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[70]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[71]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[72]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[73]" [label="VariableDeclarator\n < d >", fillcolor=white   style=filled  ]
"List[74]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[75]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[76]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[77]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[78]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[79]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[80]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[81]" [label="VariableDeclarator\n < e >", fillcolor=white   style=filled  ]
"List[82]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[83]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[84]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[85]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[86]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[87]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[88]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[89]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[90]" [label="VariableDeclarator\n < f >", fillcolor=white   style=filled  ]
"List[91]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[92]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"Modifiers[32]" -> "List[33]" [style=dashed, color=gray]
"VarDeclStmt[31]" -> "Modifiers[32]" [style=dashed, color=gray]
"VarDeclStmt[31]" -> "PrimitiveTypeAccess[34]" [style=dashed, color=gray]
"VariableDeclarator[36]" -> "List[37]" [style=dashed, color=gray]
"Opt[38]" -> "BooleanLiteral[39]" [style=dashed, color=gray]
"VariableDeclarator[36]" -> "Opt[38]" [style=dashed, color=gray]
"List[35]" -> "VariableDeclarator[36]" [style=dashed, color=gray]
"VarDeclStmt[31]" -> "List[35]" [style=dashed, color=gray]
"List[30]" -> "VarDeclStmt[31]" [style=dashed, color=gray]
"Modifiers[41]" -> "List[42]" [style=dashed, color=gray]
"VarDeclStmt[40]" -> "Modifiers[41]" [style=dashed, color=gray]
"VarDeclStmt[40]" -> "PrimitiveTypeAccess[43]" [style=dashed, color=gray]
"VariableDeclarator[45]" -> "List[46]" [style=dashed, color=gray]
"Opt[47]" -> "BooleanLiteral[48]" [style=dashed, color=gray]
"VariableDeclarator[45]" -> "Opt[47]" [style=dashed, color=gray]
"List[44]" -> "VariableDeclarator[45]" [style=dashed, color=gray]
"VarDeclStmt[40]" -> "List[44]" [style=dashed, color=gray]
"List[30]" -> "VarDeclStmt[40]" [style=dashed, color=gray]
"IfStmt[49]" -> "VarAccess[50]" [style=dashed, color=gray]
"IfStmt[53]" -> "VarAccess[54]" [style=dashed, color=gray]
"Modifiers[58]" -> "List[59]" [style=dashed, color=gray]
"VarDeclStmt[57]" -> "Modifiers[58]" [style=dashed, color=gray]
"VarDeclStmt[57]" -> "PrimitiveTypeAccess[60]" [style=dashed, color=gray]
"VariableDeclarator[62]" -> "List[63]" [style=dashed, color=gray]
"VariableDeclarator[62]" -> "Opt[64]" [style=dashed, color=gray]
"List[61]" -> "VariableDeclarator[62]" [style=dashed, color=gray]
"VarDeclStmt[57]" -> "List[61]" [style=dashed, color=gray]
"List[56]" -> "VarDeclStmt[57]" [style=dashed, color=gray]
"Block[55]" -> "List[56]" [style=dashed, color=gray]
"IfStmt[53]" -> "Block[55]" [style=dashed, color=gray]
"Modifiers[69]" -> "List[70]" [style=dashed, color=gray]
"VarDeclStmt[68]" -> "Modifiers[69]" [style=dashed, color=gray]
"VarDeclStmt[68]" -> "PrimitiveTypeAccess[71]" [style=dashed, color=gray]
"VariableDeclarator[73]" -> "List[74]" [style=dashed, color=gray]
"VariableDeclarator[73]" -> "Opt[75]" [style=dashed, color=gray]
"List[72]" -> "VariableDeclarator[73]" [style=dashed, color=gray]
"VarDeclStmt[68]" -> "List[72]" [style=dashed, color=gray]
"List[67]" -> "VarDeclStmt[68]" [style=dashed, color=gray]
"Block[66]" -> "List[67]" [style=dashed, color=gray]
"Opt[65]" -> "Block[66]" [style=dashed, color=gray]
"IfStmt[53]" -> "Opt[65]" [style=dashed, color=gray]
"List[52]" -> "IfStmt[53]" [style=dashed, color=gray]
"Modifiers[77]" -> "List[78]" [style=dashed, color=gray]
"VarDeclStmt[76]" -> "Modifiers[77]" [style=dashed, color=gray]
"VarDeclStmt[76]" -> "PrimitiveTypeAccess[79]" [style=dashed, color=gray]
"VariableDeclarator[81]" -> "List[82]" [style=dashed, color=gray]
"VariableDeclarator[81]" -> "Opt[83]" [style=dashed, color=gray]
"List[80]" -> "VariableDeclarator[81]" [style=dashed, color=gray]
"VarDeclStmt[76]" -> "List[80]" [style=dashed, color=gray]
"List[52]" -> "VarDeclStmt[76]" [style=dashed, color=gray]
"Block[51]" -> "List[52]" [style=dashed, color=gray]
"IfStmt[49]" -> "Block[51]" [style=dashed, color=gray]
"IfStmt[49]" -> "Opt[84]" [style=dashed, color=gray]
"List[30]" -> "IfStmt[49]" [style=dashed, color=gray]
"Modifiers[86]" -> "List[87]" [style=dashed, color=gray]
"VarDeclStmt[85]" -> "Modifiers[86]" [style=dashed, color=gray]
"VarDeclStmt[85]" -> "PrimitiveTypeAccess[88]" [style=dashed, color=gray]
"VariableDeclarator[90]" -> "List[91]" [style=dashed, color=gray]
"VariableDeclarator[90]" -> "Opt[92]" [style=dashed, color=gray]
"List[89]" -> "VariableDeclarator[90]" [style=dashed, color=gray]
"VarDeclStmt[85]" -> "List[89]" [style=dashed, color=gray]
"List[30]" -> "VarDeclStmt[85]" [style=dashed, color=gray]
"Block[29]" -> "List[30]" [style=dashed, color=gray]
"InstanceInitializer[28]" -> "Block[29]" [style=dashed, color=gray]
"List[27]" -> "InstanceInitializer[28]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "VariableDeclarator[90]" [color=red, constraint=false]
"VariableDeclarator[90]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[90]" -> "VarAccess[50]" [color=red, constraint=false]
"VariableDeclarator[90]" -> "VariableDeclarator[81]" [color=red, constraint=false]
"VariableDeclarator[81]" -> "VariableDeclarator[90]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[81]" -> "VariableDeclarator[62]" [color=red, constraint=false]
"VariableDeclarator[81]" -> "VariableDeclarator[73]" [color=red, constraint=false]
"VariableDeclarator[62]" -> "VariableDeclarator[81]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[62]" -> "VarAccess[54]" [color=red, constraint=false]
"VariableDeclarator[73]" -> "VariableDeclarator[81]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[73]" -> "VarAccess[54]" [color=red, constraint=false]
"VarAccess[54]" -> "VariableDeclarator[62]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[54]" -> "VariableDeclarator[73]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[54]" -> "VarAccess[50]" [color=red, constraint=false]
"VarAccess[50]" -> "VarAccess[54]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[50]" -> "VariableDeclarator[90]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[50]" -> "VariableDeclarator[45]" [color=red, constraint=false]
"VariableDeclarator[45]" -> "VarAccess[50]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[45]" -> "BooleanLiteral[48]" [color=red, constraint=false]
"BooleanLiteral[48]" -> "VariableDeclarator[45]" [color=blue, constraint=false, xlabel=" "] 
"BooleanLiteral[48]" -> "VariableDeclarator[36]" [color=red, constraint=false]
"VariableDeclarator[36]" -> "BooleanLiteral[48]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[36]" -> "BooleanLiteral[39]" [color=red, constraint=false]
"BooleanLiteral[39]" -> "VariableDeclarator[36]" [color=blue, constraint=false, xlabel=" "] 
"BooleanLiteral[39]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "BooleanLiteral[39]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"VarDeclStmt[31]" -> "VarDeclStmt[40]" -> "IfStmt[49]" -> "VarDeclStmt[85]" [style=invis]
"Modifiers[32]" -> "PrimitiveTypeAccess[34]" -> "List[35]" [style=invis]
"List[37]" -> "Opt[38]" [style=invis]
"Modifiers[41]" -> "PrimitiveTypeAccess[43]" -> "List[44]" [style=invis]
"List[46]" -> "Opt[47]" [style=invis]
"VarAccess[50]" -> "Block[51]" -> "Opt[84]" [style=invis]
"IfStmt[53]" -> "VarDeclStmt[76]" [style=invis]
"VarAccess[54]" -> "Block[55]" -> "Opt[65]" [style=invis]
"Modifiers[58]" -> "PrimitiveTypeAccess[60]" -> "List[61]" [style=invis]
"List[63]" -> "Opt[64]" [style=invis]
"Modifiers[69]" -> "PrimitiveTypeAccess[71]" -> "List[72]" [style=invis]
"List[74]" -> "Opt[75]" [style=invis]
"Modifiers[77]" -> "PrimitiveTypeAccess[79]" -> "List[80]" [style=invis]
"List[82]" -> "Opt[83]" [style=invis]
"Modifiers[86]" -> "PrimitiveTypeAccess[88]" -> "List[89]" [style=invis]
"List[91]" -> "Opt[92]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
83", fillcolor=1]
Edges [label="Total Nr. Edges
110", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
13", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
14", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}