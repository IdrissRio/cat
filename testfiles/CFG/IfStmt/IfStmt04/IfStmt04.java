public class IfStmt04 {
  {
    boolean b = true;
    boolean a = true;
    if (b) {
      if (a) {
        int c;
      } else {
        int d;
      }
      int e;
    }
    int f;
  }
}