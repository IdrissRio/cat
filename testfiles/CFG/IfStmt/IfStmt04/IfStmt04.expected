NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [BooleanLiteral[39]]
NodeName: BooleanLiteral[39] Successors: [VariableDeclarator[36]]
NodeName: VariableDeclarator[36] Successors: [BooleanLiteral[48]]
NodeName: BooleanLiteral[48] Successors: [VariableDeclarator[45]]
NodeName: VariableDeclarator[45] Successors: [VarAccess[50]]
NodeName: VarAccess[50] Successors: [VarAccess[54], VariableDeclarator[90]]
NodeName: VarAccess[54] Successors: [VariableDeclarator[62], VariableDeclarator[73]]
NodeName: VariableDeclarator[62] Successors: [VariableDeclarator[81]]
NodeName: VariableDeclarator[81] Successors: [VariableDeclarator[90]]
NodeName: VariableDeclarator[90] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: VariableDeclarator[73] Successors: [VariableDeclarator[81]]

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [VariableDeclarator[90]]
NodeName: VariableDeclarator[90] Predecessor: [VarAccess[50], VariableDeclarator[81]]
NodeName: VarAccess[50] Predecessor: [VariableDeclarator[45]]
NodeName: VariableDeclarator[45] Predecessor: [BooleanLiteral[48]]
NodeName: BooleanLiteral[48] Predecessor: [VariableDeclarator[36]]
NodeName: VariableDeclarator[36] Predecessor: [BooleanLiteral[39]]
NodeName: BooleanLiteral[39] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: VariableDeclarator[81] Predecessor: [VariableDeclarator[62], VariableDeclarator[73]]
NodeName: VariableDeclarator[62] Predecessor: [VarAccess[54]]
NodeName: VarAccess[54] Predecessor: [VarAccess[50]]
NodeName: VariableDeclarator[73] Predecessor: [VarAccess[54]]
